include/wil/resource.h

```diff
@@ -4954,6 +4956,16 @@ namespace wil
     typedef weak_any<shared_cotaskmem_string> weak_cotaskmem_string;
 #endif // __WIL_OBJBASE_H_STL
 
+#if defined(_SHLOBJ_H_) && !defined(__WIL_SHLOBJ_H__) && WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+#define __WIL_SHLOBJ_H__
+    typedef unique_any<LPITEMIDLIST, decltype(&::ILFree), ::ILFree> unique_itemidlist;

```

Comment:
ILFree() is defined to call CoTaskMemFree() so it is not needed.

Reply:
In [the documentation](https://docs.microsoft.com/en-us/windows/desktop/api/shlobj_core/nf-shlobj_core-ilfree), this is only mentioned for Windows 2000, which makes the use of CoTaskMemFree effectively implementation-defined behavior on anything more recent.

Reply:
Actually, I might just remove that entirely, since `wil::unique_cotaskmem_ptr<ITEMIDLIST_*>` can be used instead.